.github/workflows/maven.yml

```diff
@@ -26,6 +26,10 @@ permissions:
   id-token: write
   checks: write
 
+concurrency:
+  group: ${{ github.ref }}
+  cancel-in-progress: true
```

Comment:
nice, looking at https://docs.github.com/en/actions/using-jobs/using-concurrency now...

So this applies to lukasj:gh_cancel branch in this case such that if you push another commit, any previous still running test is stopped so that a new test is started.  Very good idea!